src/gas.rs

```diff
@@ -18,9 +20,9 @@ pub fn update_call_index(instructions: &mut elements::Instructions, inserted_ind
 	}
 }
 
-/// A block of code represented by it's start position and cost.
-///
-/// The block typically starts with instructions such as `loop`, `block`, `if`, etc.
+/// A control flow block is opened with the `block`, `loop`, and `if` instructions and is closed
+/// with `end`. Each block implicitly defines a new label. The control blocks form a stack during
```

Comment:
```suggestion
/// with `end` or `else`. Each block implicitly defines a new label. The control blocks form a stack during
```

Reply:
Well, as it is now, `else` is in the same control block as `if`, so it doesn't actually close a control block, just a metering block.

Reply:
OK, so I added a test case which demonstrates why `else` does not close a control block and open a new one (`branch_from_if_not_else`). Granted, there are other ways to resolve this issue but simple finalizing the metered block is the simplest IMO and makes sense to me. So I'm inclined not to change this comment.